### PR TITLE
[FIX] bank_account created by the import must belong to the partner link...

### PR DESCRIPTION
**Pull request by lmignon**
...ed to the company of the provided journal

PR on master: odoo/odoo#5821

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Thanks :+1:

**Comment by sbidoul** [comment]
I see the bug and I think your fix is correct.

But I'm not sure in which circumstances this bug is being triggered, because we don't want to autocreate company bank accounts (only counterpart bank accounts). I probably overlook something.

**Comment by hbrunn** [comment]
@lmignon is this still WIP? And could you rebase please?

**Comment by legalsylvain** [comment]
ping @lmignon

File: account_bank_statement_import/tests/test_import_bank_statement.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
+# -*- coding: utf-8 -*-
+##############################################################################
```
Comment (NL66278) [inline account_bank_statement_import/tests/test_import_bank_statement.py:2]: Please use shortened headers.

File: account_bank_statement_import/account_bank_statement_import.py [hunk 2]
```diff
@@ -198,9 +198,13 @@ def _create_bank_account(self, cr, uid, account_number, journal_id=False, partne
         # while 'counterpart' bank accounts (from which statement transactions originate) don't.
         # Warning : if company_id is set, the method post_write of class bank will create a journal
         if journal_id:
-            vals_acc['partner_id'] = uid
+            company_id = self.pool['account.journal'].browse(
```
Comment (NL66278) [inline account_bank_statement_import/account_bank_statement_import.py:201]: Why not self.env and new API?

File: account_bank_statement_import/account_bank_statement_import.py [hunk 3]
```diff
@@ -198,9 +198,13 @@ def _create_bank_account(self, cr, uid, account_number, journal_id=False, partne
...
+                cr, uid, journal_id, context=context).company_id.id
+            vals = self.pool['res.partner.bank'].onchange_company_id(
+                cr, uid, None, company_id, context=None)
+            vals_acc.update(vals.get('value', {}))
             vals_acc['journal_id'] = journal_id
-            vals_acc['company_id'] = self.pool.get('res.users').browse(cr, uid, uid, context=context).company_id.id
+            vals_acc['company_id'] = company_id
 
         return self.pool.get('res.partner.bank').create(cr, uid, vals_acc, context=context)
```
Comment (NL66278) [inline account_bank_statement_import/account_bank_statement_import.py:209]: Also here why not self.env and new API?

**Comment by NL66278** [comment]
:+1:  With minor nitpicks and agreeing with the remark that this bug will only occur in those cases where autocreate of company bank accounts is done, which normally is not the case.

**Comment by pedrobaeza** [comment]
This is conflicting, so I suppose this has been already improved/fixed meanwhile, and thus I close this. If not, please feel free to reopen fixing conflicts.

**Pull request closed without merging**